hyperspy/misc/math/filter/snip.py

```diff
@@ -0,0 +1,75 @@
+import numpy as np
+
+def snip_method(spectrum,
+                width=13, decrease_factor=np.sqrt(2),
+                iterations=16):
+    """
+    use snip algorithm to obtain background
+
+    Parameters
+    ----------
+    spectrum : array
+        intensity spectrum or array 
+    width : int, optional
+        window size in indices. Typically twice the fwhm of peaks in spectrum 
+    decrease_factor : float, optional
+        gradually decrease of window size, default as sqrt(2)
+    iterations : int, optional
+        Number of iterations in snip filter
+
+
+    Returns
+    -------
+    background : array
+        output background curve
+
+    References
+    ----------
+
+    .. [1] C.G. Ryan etc, "SNIP, a statistics-sensitive background
+           treatment for the quantitative analysis of PIXE spectra in
+           geoscience applications", Nuclear Instruments and Methods in
+           Physics Research Section B, vol. 34, 1998.
+    """
+
+    background = np.array(spectrum)
+    spectra_size = len(background)
+    window_p = int(width)
```

Comment:
It would be better to raise an error if width is a float. At the moment people can specify a float, and the code will automatically convert this. `raise ValueError('The window_p argument must be an integer')`

Reply:
also, what does the p in window_p stand for?